### PR TITLE
[WIP]error out in listing service account for non existing user

### LINKED ISSUE #14937: Service Account listing does not error out on non-existing user
## NOTE
If this case is urgent, please subscribe to [Subnet](https://min.io/pricing) so that our 24/7 support team may help you faster.

Using `mc` to list service account for a user.
For valid user `alpha`
```
$ mc admin user svcacct list myminio alpha
svcalpha            
```
For non existing user , `alpha1` in this case , there is no error . 
```
$ mc admin user svcacct list myminio alpha1
```

User present in system 
```
ashish@ashish:~/code/go/src/github/minio/mc$ ./mc admin user list myminio
enabled    alpha                 consoleAdmin        
enabled    beta                  consoleAdmin
```

## Expected Behavior

mc service account listing for non existing user must error out as shown below
```
ashish@ashish:~/code/go/src/github/minio/mc$ ./mc admin user svcacct list myminio alpha1
mc: <ERROR> Unable to list service accounts. The specified user does not exist. (Specified user does not exist).
```

## Current Behavior

No output put on console for service account list of an non existing user
```
/mc admin user svcacct list myminio alpha1
```

## Steps to Reproduce (for bugs)

1. Create a user and service account in MinIO
2. List the service account for that user. Listing is successful
3. List the service account for any arbitarry user not present in the system. Listing shows nothing . It must error out

## Regression
 No

## Your Environment

* Version used (`minio --version`): master
* Server setup and configuration:
* Operating System and version (`uname -a`):  
Linux ashish 5.13.0-40-generic #45~20.04.1-Ubuntu SMP Mon Apr 4 09:38:31 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### PR DESCRIPTION
**Pull request by sinhaashish**
## Description
Details in issue https://github.com/minio/minio/issues/14937

Fixes #14937 
## How to test this PR?

Do a service account listing on some non existing user 
It should error out as shown below
```
ashish@ashish:~/code/go/src/github/minio/mc$ ./mc admin user svcacct list myminio alpha1
mc: <ERROR> Unable to list service accounts. The specified user does not exist. (Specified user does not exist).
```

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
@sinhaashish can you take a look at the CI/CD failure?

**Comment by donatello** [comment]
@sinhaashish as mentioned offline, parent account is virtual in LDAP and openid

**Review by vadmeste — COMMENTED** [review]
Maybe you can do the same thing for ListTempAccounts(), LGTM otherwise

**Comment by vadmeste** [comment]
> @sinhaashish as mentioned offline, parent account is virtual in LDAP and openid

oh just saw this

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/iam-store.go [hunk 1]
```diff
@@ -1810,19 +1810,25 @@ func (store *IAMStoreSys) ListTempAccounts(ctx context.Context, accessKey string
 
 // ListServiceAccounts - lists only service accounts from the cache.
 func (store *IAMStoreSys) ListServiceAccounts(ctx context.Context, accessKey string) ([]auth.Credentials, error) {
-	cache := store.rlock()
```
Comment (harshavardhana) [inline cmd/iam-store.go:1813]: How will this work?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by harshavardhana** [comment]
should this PR be closed since #14977 this is merged? @donatello @sinhaashish ?

**Comment by donatello** [comment]
> should this PR be closed since #14977 this is merged? @donatello @sinhaashish ?

Yes

**Pull request closed without merging**